File: angular/src/app/_model/page-params.ts
export class PageParams {
  start: number;
  page: number;
  orderBy: string;
  orderDir: string;
  params;

  constructor(start: number, page: number, params, orderBy: string, orderDir: string){
    this.start = start;
    this.page = page;
    this.params = params;
    this.orderBy = orderBy;
    this.orderDir = orderDir;
  }
}

File: spring-base/src/main/java/ru/dfsystems/spring/tutorial/dto/PageParams.java
package ru.dfsystems.spring.tutorial.dto;

import lombok.Data;

import java.io.Serializable;

/**
 * Параметры для page
 */
@Data
public class PageParams<T> implements Serializable {
    /**
     * Начиная с какого элемента выводится на текущцю страницу
     */
    private int start;
    /**
     * Количество элементов на одной странице
     */
    private int page;
    /**
     * Параметры для конкретной сущности (RoomParams etc)
     */
    private T params;
}
